R/calcEmiTarget.R

```diff
@@ -49,18 +67,71 @@ calcEmiTarget <- function(sources, subtype, scenario) {
     )
   }
 
+  # convert target factors to magclass objects
   # ensure that all magclass objects in the list have matching years so they can be bound together
-
   listYears <- lapply(listGhgFactors, getItems, dim = "year") %>% unlist() %>% unique() %>% sort()
   ghgFactor <- purrr::map(listGhgFactors,
                           ~ add_columns(.x, listYears[!listYears %in% getItems(.x, dim = "year")], 2))
-
   ghgFactor <- mbind(ghgFactor)
   ghgFactor <- ghgFactor[, sort(getYears(ghgFactor)), ]
 
-  # GHG factors weighted by GHG in 2015
-  if (subtype == "Ghgfactor") {
 
+  # 2. Extrapolate NDC targets from 2030 to 2035 for countries which do not have 2035 NDC targets ----
+
+  # add NDC versions "Emissions_NDC " and "Emissions_2025_uncond_extrapol"
+  # that take the 2035 targets from the 2025 version, but extrapolate 2030 targets to 2035 for countries which do not have 2035 targets yet
+  # create new NDC version 2025 with extrapolated targets
+
+  # if no 2035 targets available, add 2035 to target factor object
+  if (! "y2035" %in% getYears(ghgFactor)) {
+    ghgFactor <- mbind(ghgFactor, new.magpie(getRegions(ghgFactor),"y2035",getNames(ghgFactor), fill = NA))
```

Comment:
magclass has a function for this: `ghgGactor <- add_columns(ghgFactor, addnm = "y2035", dim = 2, fill = NA)`